src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1887,5 +1892,16 @@ public String getPairedReadName() {
     public final Set<SAMFlag> getSAMFlags() {
         return SAMFlag.getFlags( this.getFlags() );
     }
+
+    /** Fetches the value of a transient attribute on the SAMRecord, of null if not set. */
+    public final Object getTransientAttribute(final Object key) {
+        return (this.transientAttributes == null) ? null : this.transientAttributes.get(key);
+    }
+
+    /** Sets the value of a transient attribute, and returns the previous value if defined. */
+    public final Object setTransientAttribute(final Object key, final Object value) {
```

Comment:
Could you add a comment about how this is intended to be used?  I.e. for caching results of expensive computations.